corehq/apps/app_manager/templates/app_manager/form_summary_base.html

```diff
@@ -65,7 +65,10 @@ <h5>
                 <ol data-bind="foreach: questions">
                     <li data-bind="visible: isVisible">
                         <i data-bind="attr: { 'class': $root.questionIcon($data), 'title': type }"></i>
+                        <!-- ko ifnot: $root.readOnly -->
                         <a data-bind="attr: { 'href': $parent.url + hashtagValue }, text: $root.showIds() ? value :  $root.translateQuestion($data)" target="_blank"></a>
+                        <!-- /ko  -->
+                        <!-- ko if: $root.readOnly --><!-- ko text: $root.showIds() ? value :  $root.translateQuestion($data) --><!-- /ko --><!-- /ko -->
```

Comment:
Does this mean knockout inserts something like `<p>value</p>` ?

Reply:
It only inserts `value` as text. I removed all the extra `<span data-bind='text: value'>` from this page as it was drastically affecting performance when there were thousands of questions displayed (and so there were hundreds of thousands of unnecessary nodes on the page).